tools/src/wyvern/tools/typedAST/core/declarations/ConstructDeclaration.java

```diff
@@ -0,0 +1,126 @@
+package wyvern.tools.typedAST.core.declarations;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import wyvern.target.corewyvernIL.FormalArg;
+import wyvern.target.corewyvernIL.decltype.DeclType;
+import wyvern.target.corewyvernIL.decltype.ValDeclType;
+import wyvern.target.corewyvernIL.expression.Variable;
+import wyvern.target.corewyvernIL.modules.TypedModuleSpec;
+import wyvern.target.corewyvernIL.support.GenContext;
+import wyvern.target.corewyvernIL.support.TopLevelContext;
+import wyvern.target.corewyvernIL.type.ValueType;
+import wyvern.tools.errors.FileLocation;
+import wyvern.tools.typedAST.abs.Declaration;
+import wyvern.tools.typedAST.core.binding.NameBinding;
+import wyvern.tools.typedAST.core.binding.NameBindingImpl;
+import wyvern.tools.typedAST.interfaces.CoreAST;
+import wyvern.tools.typedAST.interfaces.ExpressionAST;
+import wyvern.tools.typedAST.interfaces.TypedAST;
+import wyvern.tools.types.Type;
+import wyvern.tools.types.UnresolvedType;
+
+public class ConstructDeclaration extends Declaration implements CoreAST {
+    private ExpressionAST definition;
```

Comment:
There is no definition; remove this field and the corresponding constructor argument